### ISSUE TITLE
The function x**x is wrong with negative non-integers

### ISSUE THREAD

**Issue by EncryptCrystal** [numworks/epsilon#1271]
#### Describe the bug
The values in the second image did not match those in the third image.

#### Screenshots
![num](https://user-images.githubusercontent.com/52292872/72174895-1320ea00-33db-11ea-8d39-a0cf6fcdecdc.png)
![num2](https://user-images.githubusercontent.com/52292872/72175561-b7576080-33dc-11ea-9f6b-9e07327677e9.png)
![num3](https://user-images.githubusercontent.com/52292872/72175568-b8888d80-33dc-11ea-89d1-8d6347a127a0.png)

#### Expected behavior
I expected the values of the second image to match that of the third image

#### Environment
 - Actual online simulator

**Comment by jacobly0** [comment]
Yeah, a strange property of float approximation is that -10 + 0.1 may not be the same value as -9.9.  That said, I imagine the calculation app is doing symbolic computations and the graphing app is doing approximate calculations, and approximating a negative number raised to a negative number is impossible in principle given its discontinuous nature.

**Comment by nsajko** [comment]
@Ecco @LeaNumworks @EmilieNumworks @RubenNumworks 

There are two issues shown by the screenshots and reproduced in d14638163a5229488cd2e3e864892a5eb406a107

Firstly note that we are talking about _complex numbers_ here.

The more important issue is that the "Calculation" app **returns nonsensical results** for calculations like `(-9.8)**(-9.8)`: the given result equals `-‖(-9.8)**(-9.8)‖` (where `‖x‖` is the magnitude of `x`), which is **obviously a bug** as it is useless and nonsensical.

The other issue is that the "Functions" app does not handle complex numbers. Maybe it should configurably return just the real part, just the magnitude; or it could even graph both the complex and real parts? But this is probably a topic for another issue.

**Comment by nsajko** [comment]
Here are some simpler reproducers, note how the returned value for a complex number is either the magnitude or -magnitude (where the latter possibility does not make any sense):

![screenshot-0 4](https://user-images.githubusercontent.com/4944410/73703785-072dfc80-46e9-11ea-97e7-4b5f43aef924.png)
![screenshot-0 6](https://user-images.githubusercontent.com/4944410/73703788-09905680-46e9-11ea-92de-37cd49dc8420.png)

**Comment by jacobly0** [comment]
I don't understand the issue here.  The principle root is defined as the positive real if it exists, otherwise the negative real if it exists.
`(-1)^0.4 = 1`→`(-1)^2 = (1)^5`→`1=1`
`(-1)^0.5 ≠ r`→`(-1)^1 ≠ (r)^2∀r∈ℝ` because `i` and `-i` are the only roots and neither are real
`(-1)^0.6 = -1`→`(-1)^3 = (-1)^5`→`-1=-1`
And in complex mode it seems to be choosing the correct complex root too...

**Comment by nsajko** [comment]
Yeah, I was totally wrong.

**Comment by KarineNumWorks** [comment]
Hello ! this problem seems fixed in our most recent version (v24 available in beta) ! I will therefore close this issue.
